mypyc/irbuild/expression.py

```diff
@@ -559,6 +566,47 @@ def transform_basic_comparison(builder: IRBuilder,
     return target
 
 
+def translate_str_format_percent_sign(builder: IRBuilder,
+                                      format_expr: StrExpr,
+                                      rhs: Expression) -> Value:
+    literals, conversion_types = tokenizer_printf_style(format_expr.value)
+    variables = []
+    if isinstance(rhs, TupleExpr):
+        raw_variables = rhs.items
+    elif isinstance(rhs, RefExpr):
```

Comment:
Why do we need to limit ourselves to just references and not arbitrary expressions? What is wrong with `"Hello %s" % data["name"]`?